models/experimental/functional_vit/tt/ttnn_optimized_vit_highres_wh.py

```diff
@@ -0,0 +1,539 @@
+    (
+        query,
+        key,
+        value,
+    ) = ttnn.transformer.split_query_key_value_and_split_heads(
+        query_key_value,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        num_heads=num_heads,
+    )
+    ttnn.deallocate(query_key_value)
+    ttnn.reallocate(value)
+
+    attention_scores = ttnn.matmul(
+        query,
+        key,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        dtype=ttnn.bfloat8_b,
+        core_grid=ttnn.CoreGrid(y=8, x=8),
+        # program_config=program_configs["query_by_key_matmul_program_config"],
+    )
+    ttnn.deallocate(query)
+    ttnn.deallocate(key)
+    """
+    attention_probs = ttnn.transformer.attention_softmax_(
+        attention_scores,
+        attention_mask=attention_mask,
+        head_size=head_size,
+        # program_config=program_configs["softmax_program_config"],
+    )
+
+    context_layer = ttnn.matmul(
+        attention_probs,
+        value,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        dtype=ttnn.bfloat8_b,
+        core_grid=ttnn.CoreGrid(y=8, x=8),
+        # program_config=program_configs["attention_probabilities_by_value_matmul_program_config"],
+    )
+    ttnn.deallocate(attention_probs)
+    ttnn.deallocate(value)
+
+    context_layer = ttnn.transformer.concatenate_heads(
+        context_layer,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+    )
+
+    self_output = ttnn.linear(
+        context_layer,
+        parameters.output.dense.weight,
+        bias=parameters.output.dense.bias,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        dtype=ttnn.bfloat8_b,
+        core_grid=ttnn.CoreGrid(y=8, x=8),
+        # program_config=program_configs["self_output_matmul_program_config"],
+    )
+    ttnn.deallocate(context_layer)
+
+    return self_output
+    """
+    return attention_scores
+
+
+def vit_attention(
+    config,
+    hidden_states,
+    attention_mask,
+    parameters,
+):
+    num_heads = config.num_attention_heads
+    *_, hidden_size = hidden_states.shape
+    head_size = hidden_size // num_heads
+
+    query_key_value = ttnn.linear(
+        hidden_states,
+        parameters.attention.query_key_value.weight,
+        bias=parameters.attention.query_key_value.bias,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        dtype=ttnn.bfloat8_b,
+        core_grid=ttnn.CoreGrid(y=8, x=8),
+        # program_config=program_configs["query_key_value_matmul_program_config"],
+    )
+    ttnn.reallocate(hidden_states)
+    (
+        query,
+        key,
+        value,
+    ) = ttnn.transformer.split_query_key_value_and_split_heads(
+        query_key_value,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        num_heads=num_heads,
+    )
+    ttnn.deallocate(query_key_value)
+    value = ttnn.reallocate(value)
+
+    attention_scores = ttnn.matmul(
+        query,
+        key,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+        dtype=ttnn.bfloat8_b,
+        core_grid=ttnn.CoreGrid(y=8, x=8),
```

Comment:
This line (`core_grid=ttnn.CoreGrid(y=8, x=8),`) needs to be commented out or removed to not run out of memory.

Reply:
@bbradelTT Sure, I'll try removing core_grid and test the model.